cmd/kubeadm/app/cmd/join.go

```diff
@@ -162,14 +161,30 @@ func NewCmdJoin(out io.Writer, joinOptions *joinOptions) *cobra.Command {
 			c, err := joinRunner.InitData(args)
 			kubeadmutil.CheckErr(err)
 
+			data := c.(*joinData)
+			fmt.Printf("[join] Using Kubernetes version: %s\n", data.initCfg.KubernetesVersion)
+
 			err = joinRunner.Run(args)
 			kubeadmutil.CheckErr(err)
 
-			// TODO: remove this once we have all phases in place.
-			// the method joinData.Run() itself should be removed too.
-			data := c.(*joinData)
-			err = data.Run()
-			kubeadmutil.CheckErr(err)
+			// if the node is hosting a new control plane instance
+			if data.cfg.ControlPlane != nil {
+				// outputs the join control plane done message and exit
+				etcdMessage := ""
+				if data.initCfg.Etcd.External == nil {
+					etcdMessage = "* A new etcd member was added to the local/stacked etcd cluster."
+				}
+
+				ctx := map[string]string{
+					"KubeConfigPath": data.KubeConfigPath(),
+					"etcdMessage":    etcdMessage,
+				}
+				joinControPlaneDoneTemp.Execute(data.outputWriter, ctx)
+			}
```

Comment:
Please add a `return`  after executing `joinControPlaneDoneTemp`

Reply:
done!

Reply:
Let's have an `else` block for the `joinWorkerNodeDoneMsg` instead of the return here. It'll be more readable that way.